zerver/lib/test_classes.py

```diff
@@ -781,6 +781,34 @@ def assert_json_error(self, result: HttpResponse, msg: str, status_code: int=400
         """
         self.assertEqual(self.get_json_error(result, status_code=status_code), msg)
 
+    def get_all_json_errors(self, result: HttpResponse) -> List[str]:
+        """
+        Some API endpoints return multiple error messages and we need to
+        test for all the errors. Example JSON:
+        { "result":"error",
+          "msg":"This field cannot be blank.",
+          "errors":{
+            "pattern":["This field cannot be blank."],
+            "__all__":["No groups found in URL format string."]
+          }
+        }
+        """
+
+        try:
+            json = orjson.loads(result.content)
+        except Exception:  # nocoverage
+            json = {'errors': {}}
+        errors = json.get('errors')
+        error_list: List[str] = []
+        for error_type in errors:
+            error_list = error_list + errors.get(error_type)
+        return error_list
+
+    def assert_all_json_errors(self, result: HttpResponse, expected_errors: List[str]) -> None:
```

Comment:
If you’re not mutating `expected_errors`, use the immutable `Sequence` type rather than `List`.